File: backends/dotnet/src/Zappy.Adapters.GraphQL/Payloads/CartPayload.cs
using Zappy.Application;
using Zappy.Domain;

namespace Zappy.Adapters.GraphQL;

public sealed class CartPayload(Cart? cart, int? availableStock, IReadOnlyList<UserError> errors)
{
    public Cart? Cart { get; } = cart;

    public int? AvailableStock { get; } = availableStock;

    public IReadOnlyList<UserError> Errors { get; } = errors;

    public static CartPayload From(CartResult result) =>
        new(result.Cart, result.AvailableStock, result.Errors);
}

File: backends/java/zappy-adapters/src/main/java/com/zappymart/adapters/graphql/CartPayload.java
package com.zappymart.adapters.graphql;

import com.zappymart.domain.cart.Cart;
import com.zappymart.domain.shared.UserError;

import java.util.List;

public record CartPayload(Cart cart, Integer availableStock, List<UserError> errors) {
}
